XpressAI/xircuits
# ✨ Add Core Workflow Examples

# Description
This PR introduces 6 new examples, each designed to illustrate fundamental concepts and features in Xircuits. 

- `HelloXircuits`: Basic workflow that introduces parameters and nodes.
- `ArgumentParameters`: Demonstrates `Argument` parameter input in workflows.
- `CompulsoryPorts`: Shows mandatory data inputs for components.
- `DynamicPorts`: Highlights expandible inPorts in components.
- `SettingVariables`: Illustrates a method to define, set, and get variables.
- `XircuitsContext`: Explains `ctx` sharing across components.
- `ControlFlowExample`: Showcases an example of a if control flow mechanism.

This PR also
- updates `Argument` nodes from incorrectly showing an empty tooltip
- packages the examples directory into the wheel, and have the examples pulled from the wheel instead of the repository
- enable gh action for wheel building and ui tests to run on specified branches.

## References

https://github.com/XpressAI/xircuits/pull/284

## Pull Request Type

- [x] Others: Workflow Examples

## Type of Change


# Tests

Test A: Try 
1. Install xircuits from the wheel. run `xircuits examples`.
2. Verify that the `examples` directory is present in the current working directory.

Test B: Run All Workflow Examples

1. Navigate to the examples directory.
3. Try out each workflow example (HelloXircuits, ArgumentParameters, CompulsoryPorts, DynamicPorts, SettingVariables, XircuitsContext, ControlFlowExample).
4. Confirm that each example runs successfully and outputs the expected results without errors.


## Tested on?


## Notes
